crates/autopilot/src/run_loop.rs

```diff
@@ -141,6 +141,23 @@ impl RunLoop {
                 }
             };
 
+            // validate solution
+            let settlement = match Settlement::new(
+                &revealed.calldata.internalized.clone().into(),
+                self.eth.contracts().settlement_domain_separator(),
+            ) {
+                Ok(settlement) => settlement,
+                Err(err) => {
+                    tracing::error!(driver = %driver.name, ?err, "unexpected calldata format");
+                    return;
+                }
+            };
+
+            if !self.validate_settlement(&auction, &settlement).await {
+                tracing::error!(driver = %driver.name, "invalid settlement");
+                return;
```

Comment:
Shouldn't we still store all the relevant data for this competition when the solution fails some checks?